pyinfraformat/core/io.py

```diff
@@ -141,27 +140,43 @@ def from_gtk_wfs(bbox, coord_system, robust=True, maxholes=1000):
 
     x1, y1 = project_points(bbox[0], bbox[1], coord_system, "EPSG:4326")
     x2, y2 = project_points(bbox[2], bbox[3], coord_system, "EPSG:4326")
-
     x1, x2 = min((x1, x2)), max((x1, x2))
     y1, y2 = min((y1, y2)), max((y1, y2))
     bbox = [x1, y1, x2, y2]
 
-    wfs_io = wfs.getfeature(
-        typename=["Rajapinnat_GTK_Pohjatutkimukset_WFS:Pohjatutkimukset"],
-        maxfeatures=maxholes,
-        bbox=bbox,
-        outputFormat="GEOJSON",
-    )
-
-    data = wfs_io.read().decode("utf-8")
-    data = data.replace("\\", r"\\")
-    data_json = json.loads(data, strict=False)
-    if "features" not in data_json:
-        return Holes()
-
     results = []
-    for line in data_json["features"]:
-        results.append(line)
+    page_size = 1000
+    for startindex in range(0, int(1e100), page_size):
+        if len(results) > maxholes:
+            break
+        wfs_io = wfs.getfeature(
+            typename=["Rajapinnat_GTK_Pohjatutkimukset_WFS:Pohjatutkimukset"],
+            bbox=bbox,
+            maxfeatures=page_size,
+            startindex=startindex,
+            outputFormat="GEOJSON",
+        )
+        data = wfs_io.read().decode("utf-8")
+        data = data.replace("\\", r"\\")
+        while True:
+            try:
+                data_json = json.loads(data, strict=False)
+                break
+            except json.JSONDecodeError as error:
+                if error.msg == "Expecting ',' delimiter":
+                    msg = error.msg + " at pos " + str(error.pos)
+                    msg += """. Asuming invalid char, replacing " with '."""
+                    logger.warning(msg)
+                    data = data[: error.pos - 1] + "'" + data[error.pos :]
```

Comment:
This gets the job done, but is there a more efficient way to do this?

Reply:
Don't know/didn't find other way to do it. Another PR maybe later